Duplicati/Library/Snapshots/lvm-scripts/find-volume.sh

```diff
@@ -20,33 +20,56 @@ NAME=$1
 # Get the reported mount point for the current folder
 #
 VOLUME=`df -P "$NAME" | tail -1 | awk '{ print $1}'`
-if [ "$?" -ne 0 ]
+if [ "$?" -ne 0 ] || [ "$VOLUME" == "" ]
 then
-	EXIT_CODE=$?
+	[[ "$?" -ne 0 ]] && EXIT_CODE=$? || EXIT_CODE=-1
 	echo "Error: unable to determine device for $NAME"
 	exit $EXIT_CODE
 fi
 
 MOUNTPOINT=`df -P "$NAME" | tail -1 | awk '{ print $NF}'`
-if [ "$?" -ne 0 ]
+if [ "$?" -ne 0 ] || [ "MOUNTPOINT" == "" ]
 then
-	EXIT_CODE=$?
+	[[ "$?" -ne 0 ]] && EXIT_CODE=$? || EXIT_CODE=-1
 	echo "Error: unable to determine mount point for $NAME"
 	exit $EXIT_CODE
 fi
 
 #
 # Get the LVM path for the mapped volume
 #
-LVMID=`lvs "$VOLUME" --options vg_name,lv_name --noheadings | tail -1 | awk '{ print $1 "/" $2}'`
-if [ "$?" -ne 0 ]
+
+function get_lvmid {
+	LVMID=`lvs "$1" --options vg_name,lv_name --noheadings 2>/dev/null | tail -1 | awk '{ print $1 "/" $2}'`
+	if [ "$?" -ne 0 ] 
+	then
+		EXIT_CODE=$?
+		echo "Error: Unable to determine volume group (VG) for mapped volume $VOLUME"
+		exit $EXIT_CODE
```

Comment:
No, lvs returns non-zero value, but last pipe command "overrides" this. For example, the next bash command "`asdasdasdasd | cat`" will have zero exit code because "cat" return zero value.